plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -822,12 +823,12 @@ private boolean indexExists(MongoCollection<Document> schemaCollection)
                 .anyMatch(index -> index.getKeys().size() == 1 && TABLE_NAME_KEY.equals(index.getKeys().get(0).getName()));
     }
 
-    private Set<String> getTableMetadataNames(String schemaName)
+    private Set<SchemaTableName> getTableMetadataNames(String schemaName)
     {
         try (MongoCursor<Document> cursor = client.getDatabase(schemaName).getCollection(schemaCollection)
                 .find().projection(new Document(TABLE_NAME_KEY, true)).iterator()) {
             return Streams.stream(cursor)
-                    .map(document -> document.getString(TABLE_NAME_KEY))
+                    .map(document -> new SchemaTableName(schemaName, document.getString(TABLE_NAME_KEY)))
```

Comment:
Instead of relying on `SchemaTableName to perform the lower case - can we perform it locally here ? So it would help us when we fixing the case sensitive identifier support.